src/CryptState.cpp

```diff
@@ -55,20 +62,48 @@ void CryptState::genKey() {
 	bInit = true;
 }
 
-void CryptState::setKey(const unsigned char *rkey, const unsigned char *eiv, const unsigned char *div) {
-	memcpy(raw_key, rkey, AES_KEY_SIZE_BYTES);
-	memcpy(encrypt_iv, eiv, AES_BLOCK_SIZE);
-	memcpy(decrypt_iv, div, AES_BLOCK_SIZE);
-	AES_set_encrypt_key(raw_key, AES_KEY_SIZE_BITS, &encrypt_key);
-	AES_set_decrypt_key(raw_key, AES_KEY_SIZE_BITS, &decrypt_key);
-	bInit = true;
+void CryptStateOCB2::setKey(const std::string& rkey, const std::string& eiv, const std::string& div) {
+	if(rkey.length() == AES_KEY_SIZE_BYTES && eiv.length() == AES_BLOCK_SIZE && div.length() == AES_BLOCK_SIZE) {
```

Comment:
I assume that if this condition is not met, it's an error?

In that case this error should at least be logged (`qCritical` is your friend here).

For that you can invert the condition for the if-statement and then inside the if, log the error and exit the function

Reply:
The same applies to all other functions that only do something if a condition is met.

In fact if these are like absolutely necessary pre-conditions I think we can also use `Q_ASSERT` and let the program crash if these pre-conditions are not met.

Reply:
In the old code, if these requirements don't meet, it will... do nothing. I'm not really feeling good to just ignore them as well. I will put some `qCritical` here.

Reply:
@Avatat do these methods make sense if the conditions in the `if` statements are not fulfilled? what do you think? crash or log if the conditions aren't met?